src/Data/Entity.php

```diff
@@ -3,52 +3,60 @@
 
 abstract class Entity implements \JsonSerializable
 {
-    protected $properties = [];
-
     protected $id;
+    protected $createdAt;
+    protected $updatedAt;
+    protected $deleted = false;
 
-    private $data;
-
-    private $callMethods = ['get', 'set'];
+    public function getId()
+    {
+        return $this->id;
+    }
 
-    public function __construct($data = [])
+    public function setId($id)
```

Comment:
In my opinion, entities should not have setter methods at all. They should follow this pattern to maintain immutability:

```php
public function __construct(array $data = [])
{
    // set any property available
}

public function withData(array $data)
{
    $entity = clone $this;
    // set any properties available on cloned $entity
   return $entity;
}
```